pytorch_toolkit/ote/ote/modules/config_transformers/config_transformers_engine.py

```diff
@@ -0,0 +1,103 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import logging
+import os
+import yaml
+import datetime
+from tempfile import NamedTemporaryFile
+from copy import copy
+
+from ote.utils import load_config
+from ..builder import build_config_transformer
+
+def save_config(config, file_path):
+    with open(file_path, 'w') as output_stream:
+        yaml.dump(config, output_stream)
+
+class _ConfigTransformersHandler:
+    def __init__(self, template_path, config_transformers_names):
+        self.template_path = template_path
+        self.timestamp = self._generate_timestamp_suffix()
+
+        if config_transformers_names is None:
+            self.config_transformers = None
+            return
+
+        if not isinstance(config_transformers_names, list):
+            config_transformers_names = [config_transformers_names]
+        self.config_transformers = []
+        for cfg_transform_name in config_transformers_names:
+            self.config_transformers.append(build_config_transformer(cfg_transform_name))
```

Comment:
It looks like we should place all such functionality into `build_config_transformer`. If you want to support list of transforms you should create some class agregator and move the code from `__call__` method here. In this way no need in `_ConfigTransformersHandler` and `ConfigTransformersEngine` classes - it also preserves the regular structure in `train.py`, `eval.py` and etc, scripts.

Reply:
Done